db/config.cc

```diff
@@ -427,6 +427,8 @@ db::config::config(std::shared_ptr<db::extensions> exts)
     /* Note: Unused. Retained for upgrade compat. Deprecate and remove in a cycle or two. */
     , commitlog_reuse_segments(this, "commitlog_reuse_segments", value_status::Unused, true,
         "Whether or not to re-use commitlog segments when finished instead of deleting them. Can improve commitlog latency on some file systems.\n")        
+    , commitlog_flush_threshold_in_mb(this, "commitlog_flush_threshold_in_mb", value_status::Used, -1,
+        "Threshold for commitlog disk usage. When used disk space goes above this value, Scylla initiates flushes of memtables to disk for the oldest commitlog segments, removing those log segments. Adjusting this affects disk usage vs. write latency. Default is per-shard size minus approximately one segment size.")
```

Comment:
Segment size is controlled by `commitlog_segment_size_in_mb`. The parameter is per node, but the default is calculated to be roughly "<shard max disk footprint>- <one segment>". Avi wanted a mention of defaults, but they don't exactly roll of the tongue...

Reply:
Ok, so if I understand you now, the sentence "Default is per-shard size minus approximately one segment size." is trying to explain what the default is. But what is it saying? What is "per-shard size" referring to? Is this another parameter defined somewhere? Maybe use those parameter names explicitly instead of English that a stupid read (like myself) won't know what they refer to?

Reply:
I've updated it to just use the config parameter names. Maybe more parse-able now.

Reply:
It should be based on per-node limit, people don't normalize to shards. e.g. if system RAM is 100GB, commitlog total space should be 100GB, and the flush threshold be 100GB-nr_shards*segment_size. I'd describe it as "just under the commitlog space limit"; if we have 100 shards and 1GB/shard, it doesn't matter to the user if the limit is 1024 MiB or 992 MiB.

Separately, we need to adjust it to be half the allocated space (per-node when exposed to the user, per-shard internally). We have no reason to expect that we can flush a commitlog's worth of data in the time it takes to fill one segment.

Later we can try to make it smarter by considering the actual used space of the memtables. Something like (1 - (largest memtables size) / (space allocated for memtables)) * (total commitlog space). Note it evaluates to 1/2 when we have one table, since the replica will allow a memtable to at most half the memory allocated to memtables.

Reply:
Per @nyh:s request I changed the text to "Default is (approximately) commitlog_total_space_in_mb - <num shards>*commitlog_segment_size_in_mb". Which is what I meant and what is (roughly) is. (Some rounding and taking to small total sizes into account for tests).